### PR TITLE
Add `Circle.intersect()`

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
This PR adds the general purpose `Circle.intersect()` function. It's a function that finds all intersection points (as long as there's a finite number of them) between the circle and any other shape. Since this is a first implementation it only supports circle-circle intesection but it will support rects/lines/polygons in the future too.

A little demo to test this:
```Py
from pygame.geometry import Circle
import pygame

sw, sh = 600, 600
swh, shh = sw / 2, sh / 2
c = Circle((swh, shh), 100)

pygame.init()
screen = pygame.display.set_mode((sw, sh))
clock = pygame.time.Clock()
keep = True

c2 = Circle((swh + 50, shh + 50), 100)

angle = 1

while keep:
    clock.tick_busy_loop(60)

    screen.fill((0, 0, 0))

    c2.center = pygame.mouse.get_pos()

    pygame.draw.circle(screen, (255, 255, 255), c.center, c.r, 1)
    pygame.draw.circle(screen, (255, 255, 255), c2.center, c2.r, 1)

    for center in c.intersect(c2):
        pygame.draw.circle(screen, (255, 0, 0), center, 5)

    pygame.display.flip()

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            keep = False


pygame.quit()
```

## Credits

**Geometry Project:**
For code, docs and tests:
@novialriptide @Emc2356 @itzpr3d4t0r @ScriptLineStudios @avaxar @Matiiss @newpaxonian @maqa41 @blankRiot96
Also thanks to @Starbuck5 for kickstarting the idea and the occasional help!

**Functionality added in this PR**
- `circle.c`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/circle.c
Credits to @Emc2356 @itzpr3d4t0r @novialriptide
- `geometry.pyi`
https://github.com/pygame-community/pygame-geometry/blame/main/geometry.pyi
Credits to @Emc2356 @itzpr3d4t0r @novialriptide @ScriptLineStudios @avaxar 
- `geometry_test.py`
https://github.com/pygame-community/pygame-geometry/blame/main/test/test_circle.py
Credits to @itzpr3d4t0r @novialriptide
- `geometry.rst`
https://github.com/pygame-community/pygame-geometry/blame/main/docs/circle.rst
Credits to @itzpr3d4t0r
- `geometry.h`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/include/geometry.h
Credits to @Emc2356 @itzpr3d4t0r @maqa41

### CODE REVIEW

**Comment by damusss** [comment]
Any reason why it doesn't support Rects and FRects (as they exist already)?

**Comment by itzpr3d4t0r** [comment]
> Any reason why it doesn't support Rects and FRects (as they exist already)?

I did a direct port from the geometry repo, which doesn’t have rectangle intersection implemented yet. We agreed to implement features there first to ensure proper code attribution.

**Review by damusss — APPROVED** [review]
I tested locally and the intersection cases worked out great (as expected since this is a direct port), so the PR LGTM. One step closer for Circle to be stable (allowing the other shapes to be ported aswell)

File: buildconfig/stubs/pygame/geometry.pyi [hunk 1]
```diff
@@ -4,6 +4,8 @@ from typing import (
     Callable,
     Protocol,
     Tuple,
+    Sequence,
```
Comment (ankith26) [inline buildconfig/stubs/pygame/geometry.pyi:7]: This seems like an unneeded import

**Review by bilhox — APPROVED** [review]
LGTM !

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada:

**Pull request merged**